### PR TITLE
Rework CI to handle combined website and docs

### PR DESCRIPTION
**Pull request by meain**
## Description

This reworks CI now that we have merged docs and website into a single deployment.

## Type of change

- [ ] :sunflower: Feature
- [ ] :bug: Bugfix
- [ ] :world_map: Documentation
- [ ] :robot: Test
- [x] :computer: CI/Deployment
- [ ] :hamster: Trivial/Minor

## Issue(s)

* Fixes https://github.com/alcionai/corso/issues/1551

## Test Plan

- [x] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

**Comment by meain** [comment]
Currently I have set the s3 and cloudfront creds to `DOCS_S3_BUCKET` and `DOCS_CF_DISTRIBUTION` as the website related creds are not available in GitHub. We should create two new creds `WEBSITE_S3_BUCKET` and `WEBSITE_CF_DISTRIBUTION` in both `Testing` and Production` environment. I'll make the necessary changes once done. Other than that this PR should be ready. *Keeping it in draft so as to not merge it accidentally before we make this change.*

I'll need help in adding the new secrets as I do not have access to edit those.
/cc @vkamra

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -298,7 +298,7 @@ jobs:
         run: |
           printf 'User-agent: *\nDisallow: /' > build/robots.txt
 
-      - name: Push docs
+      - name: Push website
         run: |
           aws s3 sync build "s3://${{ secrets.DOCS_S3_BUCKET }}"
```
Comment (ntolia) [inline .github/workflows/ci.yml:303]: Can you add a ` --exclude ".git/*" --delete` here?

File: .github/workflows/website-publish.yml [hunk 2]
```diff
@@ -9,27 +9,112 @@ permissions:
...
+      - name: Install dependencies for website lint
+        run: |
+          wget https://github.com/errata-ai/vale/releases/download/v2.20.2/vale_2.20.2_Linux_64-bit.tar.gz # NOTE: update in Dockerfile when updating
+          mkdir bin && tar -xvzf vale_2.20.2_Linux_64-bit.tar.gz -C bin
+          echo "$PWD/bin" >> $GITHUB_PATH
+          npm i -g markdownlint-cli@0.32.2 # NOTE: update in Dockerfile when updating
+
+      - name: Run website lint
+        env:
+          CORSO_USE_DOCKER: -1 # prevent using docker inside makefile
```
Comment (ntolia) [inline .github/workflows/website-publish.yml]: Is this going away?

File: .github/workflows/website-publish.yml [hunk 3]
```diff
@@ -9,27 +9,112 @@ permissions:
...
         with:
           role-to-assume: ${{ secrets.AWS_IAM_ROLE }}
           role-session-name: integration-testing
           aws-region: us-east-1
 
-      - name: Build docs image
+      - name: Push website
         run: |
-          make buildimage
+          aws s3 sync build "s3://${{ secrets.DOCS_S3_BUCKET }}"
```
Comment (ntolia) [inline .github/workflows/website-publish.yml:114]: Please add ` --exclude ".git/*" --delete`

**Review by ntolia — COMMENTED** [review]
Some minor changes. One of the comments might be obsolete as I was think it was handled in a follow-on patch to this PR>

File: .github/workflows/ci.yml (hunk 1 above)
Comment (meain) [reply .github/workflows/ci.yml:303]: We are just syncing the `build` dir and `.git` will be outside that when syncing.

```
.
├── .git
└── website
    ├── build  <-- this
    ├── docs
    └── src
```

Let me add `--delete` though.

File: .github/workflows/website-publish.yml (hunk 3 above)
Comment (meain) [reply .github/workflows/website-publish.yml:114]: Same as https://github.com/alcionai/corso/pull/1568#discussion_r1028819599

File: .github/workflows/website-publish.yml (hunk 2 above)
Comment (meain) [reply .github/workflows/website-publish.yml]: Looks like I have already removed it, this file seems to be pointing to an earlier commit within the PR.

**Comment by ntolia** [comment]
@meain What is the status of this PR?

**Comment by meain** [comment]
@ntolia  The code changes are good to go. I need the GitHub secrets to be setup by someone who has access though as mentioned [here](https://github.com/alcionai/corso/pull/1568#issuecomment-1322001045). Let me bring this up in slack as well.
/cc @vkamra

**Comment by ntolia** [comment]
Can you please remove the draft status on the PR? We should try and get this in soon as we will be  publishing new blog posts shortly.

**Comment by ntolia** [comment]
Please also rebase or merge changes from main.

**Review by ntolia — COMMENTED** [review]
This looks good to me but will let @vkamra approve. @uncledru might also be interested.

**Comment by uncledru** [comment]
@meain could we go ahead and update the dependabot.yml to remove the `/docs` directory as part of the config?

**Review by vkamra — APPROVED** [review]

**Comment by meain** [comment]
Need the required checks to be changed from `Docs-Linting` and `Publishs-Docs` to `Website-Linting` and `Publish-Website`.

/cc @vkamra @uncledru

**Pull request merged**